packages/api/cms-api/src/dam/files/files.service.ts

```diff
@@ -385,6 +401,180 @@ export class FilesService {
+            id: ignoreId,
+            createdAt: ignoreCreatedAt,
+            updatedAt: ignoreUpdatedAt,
+            folder: ignoreFolder,
+            image: ignoreImage,
+            scope: ignoreScope,
+            copies: ignoreCopies,
+            ...fileProps
+        } = file;
+
+        const fileInput: CreateFileInput & { copyOf: FileInterface } = {
+            ...Utils.copy(fileProps),
+            image: fileImageInput,
+            folderId: targetFolder.id,
+            copyOf: file,
+            scope: targetFolder.scope,
+        };
+
+        return this.create(fileInput);
+    }
+
+    async copyFilesToScope({
+        user,
+        fileIds,
+        targetScope,
+        targetFolderId,
+    }: {
+        user: CurrentUserInterface;
+        fileIds: string[];
+        targetScope: DamScopeInterface;
+        targetFolderId?: string;
+    }) {
+        const getUniqueFileScopes = (files: FileInterface[]): DamScopeInterface[] => {
+            const fileScopes: DamScopeInterface[] = [];
+            for (const file of files) {
+                if (file.scope === undefined) {
+                    continue;
+                }
+
+                const isDuplicateScope = Boolean(fileScopes.find((scope) => this.contentScopeService.scopesAreEqual(scope, file.scope)));
+                if (!isDuplicateScope) {
+                    fileScopes.push(file.scope);
+                }
+            }
+            return fileScopes;
+        };
+
+        const canAccessFileScopes = (fileScopes: DamScopeInterface[], user: CurrentUserInterface) => {
+            for (const scope of fileScopes) {
+                const canAccessFileScope = this.contentScopeService.canAccessScope(scope, user);
+                if (!canAccessFileScope) {
+                    return false;
+                }
+            }
+            return true;
+        };
+
+        const getOrCreateTargetFolder = async ({
+            targetFolderId,
+            targetScope,
+            fileScopes,
+        }: {
+            targetFolderId?: string;
+            targetScope: DamScopeInterface;
+            fileScopes: DamScopeInterface[];
+        }) => {
+            let createdTargetFolderAutomatically = false;
+            let targetFolder: FolderInterface;
+            if (targetFolderId) {
+                const folder = await this.foldersService.findOneById(targetFolderId);
+                if (folder === null) {
+                    throw new Error("Specified target folder doesn't exist.");
+                }
+
+                targetFolder = folder;
+            } else {
+                const scopeString = fileScopes.length === 0 ? "unknown" : fileScopes.map((scope) => Object.values(scope).join("-")).join(", ");
+                const date = new Date();
+
+                targetFolder = await this.foldersService.create(
+                    {
+                        name: `Copy from ${scopeString} ${format(date, "dd.MM.yyyy")}, ${format(date, "HH:mm:ss")}`,
```

Comment:
One downside of creating the folder in the API is that a) the label is not translated and b) we're not using the scope labels, but the values. Do you think this might be a problem?

Reply:
I don't consider this a huge problem as the scope values are usually the lowercase labels or some abbreviation, so still understandable for the user. 

However, @nsams already moved the folder creation to the admin in the follow-up PR https://github.com/vivid-planet/comet/pull/1279/files, so it doesn't really matter.